### ISSUE TITLE
Integrate Logging Library

### ISSUE THREAD

**Issue by YashKumarVerma** [camicroscope/Caracal#76]
**Is your feature request related to a problem? Please describe.**
- the codebase currently relies on console.log for logging
- this does not provide the features like assigning context to logs and dumping them to files.
- the [readme in distro](https://www.cbsnews.com/news/scientists-335-million-seed-sperm-egg-samples-moon-noahs-ark/) says
  > Logging - Container Logging is, for HIPAA reasons, disabled. Feel free to use a different logging engine if desired, especially for development.
  which means that we could a full-feature logger for development purposes.

**Describe the solution you'd like**
- integration of a good logging service (like [morgan](https://www.npmjs.com/package/morgan)?) to aid development lifecycle.
- config to not produce logs in prod mode but give extensive logs in dev mode. Can be easily done using logging levels.

**Describe alternatives you've considered**
- any other login library?
- a custom-written bunch of methods if a dedicated library brings lot of overhead.

**Additional context**
- this would greatly ease development / debugging.

**Comment by birm** [comment]
I think you've given the wrong link to distro.
In any case, that statement on logging is no longer accurate. I've just [updated it in development](https://github.com/camicroscope/Distro/commit/4ccaccacbe200d4cfc350aeb3ff8c9607e4c943c).
In short, it turns out that very few people had concerns about PHI in the logs, but people did have issues with the mongo service (usually) not working without any hints.

Anyway, to the topic of your actual request:
I think improving logs, with or without middleware, is a good idea. We'll certainly open a reasonable PR about that on this repo. Thank you for the suggestion in any case!

**Comment by YashKumarVerma** [comment]
thanks for the detailed answer! just fixed the link 🤦🏼 

I'm interested in working on this as I'd be able to understand the codebase better. Could I?

**Comment by YashKumarVerma** [comment]
@birm ping 👀

**Comment by birm** [comment]
Of course! I may or may not be picky on the PR, but we'll always welcome improvements! Thank you for your interest.

**Comment by Parth-Thummar** [comment]
Can I implement logger module with singleton design pattern and create separate function to log different errors and info logs?
